src/main/java/com/bettercloud/vault/api/Logical.java

```diff
@@ -162,4 +150,29 @@ public LogicalResponse write(final String path, final Map<String, String> nameVa
         }
     }
 
+    private LogicalResponse buildResponseFromContent(RestResponse restResponse, int retryCount) throws VaultException {
```

Comment:
Sure, but that's just to build the return object for the calling method.  It's never actually used for any decision-making within the thread.

Perhaps a minor nitpick (hope it doesn't sound like I don't appreciate the PR!)... but whether it's in the PR or a post-merge adjustment, I'd have this method return only the `data` Map and let the calling method construct the `LogicalResponse` return type.